plugins/processors/awsapplicationsignals/internal/cardinalitycontrol/metrics_limiter.go

```diff
@@ -156,23 +151,15 @@ func (m *MetricsLimiter) filterAWSDeclaredAttributes(attributes, resourceAttribu
 }
 
 func (m *MetricsLimiter) removeStaleServices() {
-	var svcToRemove []string
-	for name, svc := range m.services {
-		if svc.rotations > 3 {
-			if svc.countSnapshot[0] == svc.countSnapshot[1] && svc.countSnapshot[1] == svc.countSnapshot[2] {
-				svc.cancelFunc()
-				svcToRemove = append(svcToRemove, name)
-			}
+	m.services.Range(func(key, value any) bool {
+		svc := value.(*service)
```

Comment:
```suggestion
		svc, ok := value.(*service)
                if !ok {
                    return true
                }
```
assertion check needed?